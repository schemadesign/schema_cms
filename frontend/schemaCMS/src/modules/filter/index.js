export { FilterRoutines } from './filter.redux';

export * from './filter.selectors';
