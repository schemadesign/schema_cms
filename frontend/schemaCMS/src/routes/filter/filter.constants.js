import { FormattedMessage } from 'react-intl';
import React from 'react';

import messages from './filter.messages';
import { LINK_ITEM } from '../../shared/components/menu/mobileMenu/mobileMenu.constants';
import { ROLES } from '../../modules/userProfile/userProfile.constants';

export const PROJECTS_ID = 'projectsNavBtn';

export const FILTER_MENU_OPTIONS = [
  {
    label: <FormattedMessage {...messages.projectsListPage} />,
    to: '/project/',
    id: PROJECTS_ID,
    type: LINK_ITEM,
    allowedRoles: [ROLES.ADMIN, ROLES.EDITOR],
  },
];
