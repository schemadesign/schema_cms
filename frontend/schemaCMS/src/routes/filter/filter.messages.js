/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  subTitle: {
    id: 'filter.subTitle',
    defaultMessage: 'Edit Filter',
  },
  projectsListPage: {
    id: 'filter.projectsListPage',
    defaultMessage: 'Projects',
  },
});
