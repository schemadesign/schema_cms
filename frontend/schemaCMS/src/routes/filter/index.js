export { Filter } from './filter.component';
