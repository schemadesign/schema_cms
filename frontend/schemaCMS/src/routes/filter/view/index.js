export { default as View } from './view.container';
