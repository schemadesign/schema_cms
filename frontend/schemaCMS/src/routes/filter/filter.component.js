import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { always, path } from 'ramda';
import { FormattedMessage } from 'react-intl';

import { Container } from './filter.styles';
import messages from './filter.messages';
import { renderWhenTrueOtherwise } from '../../shared/utils/rendering';
import { Loader } from '../../shared/components/loader';
import { FilterForm } from '../../shared/components/filterForm';
import { TopHeader } from '../../shared/components/topHeader';

export class Filter extends PureComponent {
  static propTypes = {
    updateFilter: PropTypes.func.isRequired,
    fetchFieldsInfo: PropTypes.func.isRequired,
    fieldsInfo: PropTypes.object.isRequired,
    fetchFilter: PropTypes.func.isRequired,
    removeFilter: PropTypes.func.isRequired,
    filter: PropTypes.object.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        filterId: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  state = {
    loading: true,
  };

  async componentDidMount() {
    const filterId = path(['match', 'params', 'filterId'], this.props);
    const data = await this.props.fetchFilter({ filterId });
    await this.props.fetchFieldsInfo({ dataSourceId: data.datasource.id });

    this.setState({ loading: false });
  }

  getHeaderAndMenuConfig = () => ({
    headerTitle: this.props.filter.datasource.name,
    headerSubtitle: <FormattedMessage {...messages.subTitle} />,
  });

  renderContent = renderWhenTrueOtherwise(always(<Loader />), () => (
    <Fragment>
      <TopHeader {...this.getHeaderAndMenuConfig()} />
      <FilterForm
        fieldsInfo={this.props.fieldsInfo}
        updateFilter={this.props.updateFilter}
        filter={this.props.filter}
        removeFilter={this.props.removeFilter}
      />
    </Fragment>
  ));

  render() {
    return <Container>{this.renderContent(this.state.loading)}</Container>;
  }
}
