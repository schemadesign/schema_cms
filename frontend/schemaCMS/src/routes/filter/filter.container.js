import { connect } from 'react-redux';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';

import { Filter } from './filter.component';
import { DataSourceRoutines, selectFieldsInfo } from '../../modules/dataSource';
import { FilterRoutines, selectFilter } from '../../modules/filter';

const mapStateToProps = createStructuredSelector({
  fieldsInfo: selectFieldsInfo,
  filter: selectFilter,
});

export const mapDispatchToProps = dispatch =>
  bindPromiseCreators(
    {
      fetchFieldsInfo: promisifyRoutine(DataSourceRoutines.fetchFieldsInfo),
      fetchFilter: promisifyRoutine(FilterRoutines.fetchFilter),
      updateFilter: promisifyRoutine(FilterRoutines.updateFilter),
      removeFilter: promisifyRoutine(FilterRoutines.removeFilter),
    },
    dispatch
  );

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(Filter);
