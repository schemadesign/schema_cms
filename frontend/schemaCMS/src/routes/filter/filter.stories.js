import React from 'react';
import { storiesOf } from '@storybook/react';

import { Filter } from './filter.component';
import { withTheme } from '../../.storybook/decorators';
import { history } from '../../.storybook/helpers';

export const defaultProps = {
  updateFilter: Function.prototype,
  fetchFieldsInfo: Function.prototype,
  fieldsInfo: {
    field: {
      fieldType: 'fieldType',
      filterType: ['Filter Type 1', 'Filter Type 2', 'Filter Type 3'],
      unique: 'unique',
    },
  },
  fetchFilter: () => Promise.resolve({ datasource: { id: '1' } }),
  removeFilter: Function.prototype,
  filter: {
    datasource: {
      id: '1',
      name: 'name',
    },
  },
  history,
  match: {
    params: {
      filterId: '1',
      step: '5',
    },
  },
};

storiesOf('Filter', module)
  .addDecorator(withTheme())
  .add('Default', () => <Filter {...defaultProps} />);
