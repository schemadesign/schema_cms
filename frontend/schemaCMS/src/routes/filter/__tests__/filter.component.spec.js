import React from 'react';
import { shallow } from 'enzyme';

import { Filter } from '../filter.component';
import { defaultProps } from '../filter.stories';

describe('Filter: Component', () => {
  const component = props => <Filter {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly with loader', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', async () => {
    defaultProps.fetchFilter = jest.fn().mockReturnValue(Promise.resolve({ datasource: '1' }));
    defaultProps.fetchFieldsInfo = jest.fn().mockReturnValue(Promise.resolve({}));
    const wrapper = render(defaultProps);
    await Promise.resolve();
    await Promise.resolve();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchFilter on componentDidMount', async () => {
    jest.spyOn(defaultProps, 'fetchFilter');

    await render();

    expect(defaultProps.fetchFilter).toHaveBeenCalled();
  });

  it('should call fetchFilter on componentDidMount', async () => {
    defaultProps.fetchFilter = jest.fn().mockReturnValue(Promise.resolve({ datasource: '1' }));
    jest.spyOn(defaultProps, 'fetchFieldsInfo');

    await render();

    expect(defaultProps.fetchFieldsInfo).toHaveBeenCalled();
  });
});
