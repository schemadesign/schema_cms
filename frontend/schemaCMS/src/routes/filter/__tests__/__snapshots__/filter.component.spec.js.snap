// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter: Component should render correctly 1`] = `
<styled.div>
  <MobileMenu
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Edit Filter"
        id="filter.subTitle"
        values={Object {}}
      />
    }
    headerTitle="name"
    options={
      Array [
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Projects"
            id="filter.projectsListPage"
            values={Object {}}
          />,
          "to": "/project/",
          "type": "link_item",
        },
      ]
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Edit Filter"
        id="filter.subTitle"
        values={Object {}}
      />
    }
    title="name"
  />
  <LoadingWrapper
    error={null}
    loading={false}
    noData={false}
    noDataContent={null}
  >
    <FilterForm
      dataSourceId="1"
      fieldsInfo={
        Object {
          "field": Object {
            "fieldType": "fieldType",
            "filterType": Array [
              "Filter Type 1",
              "Filter Type 2",
              "Filter Type 3",
            ],
            "unique": "unique",
          },
        }
      }
      filter={
        Object {
          "datasource": Object {
            "id": "1",
            "name": "name",
          },
        }
      }
      history={
        Object {
          "goBack": [Function],
          "push": [Function],
        }
      }
      removeFilter={[Function]}
      updateFilter={[Function]}
    />
  </LoadingWrapper>
</styled.div>
`;

exports[`Filter: Component should render correctly with loader 1`] = `
<styled.div>
  <MobileMenu
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Edit Filter"
        id="filter.subTitle"
        values={Object {}}
      />
    }
    headerTitle="name"
    options={
      Array [
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Projects"
            id="filter.projectsListPage"
            values={Object {}}
          />,
          "to": "/project/",
          "type": "link_item",
        },
      ]
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Edit Filter"
        id="filter.subTitle"
        values={Object {}}
      />
    }
    title="name"
  />
  <LoadingWrapper
    error={null}
    loading={true}
    noData={false}
    noDataContent={null}
  >
    <FilterForm
      dataSourceId="1"
      fieldsInfo={
        Object {
          "field": Object {
            "fieldType": "fieldType",
            "filterType": Array [
              "Filter Type 1",
              "Filter Type 2",
              "Filter Type 3",
            ],
            "unique": "unique",
          },
        }
      }
      filter={
        Object {
          "datasource": Object {
            "id": "1",
            "name": "name",
          },
        }
      }
      history={
        Object {
          "goBack": [Function],
          "push": [Function],
        }
      }
      removeFilter={[Function]}
      updateFilter={[Function]}
    />
  </LoadingWrapper>
</styled.div>
`;
